public/js/icinga/behavior/collapsible.js

```diff
@@ -0,0 +1,216 @@
+/*! Icinga Web 2 | (c) 2019 Icinga GmbH | GPLv2+ */
+
+;(function(Icinga, $) {
+
+    'use strict';
+
+    Icinga.Behaviors = Icinga.Behaviors || {};
+
+    /**
+     * Behavior for collapsible containers.
+     *
+     * @param  icinga  Icinga  The current Icinga Object
+     */
+    var Collapsible = function(icinga) {
+        Icinga.EventListener.call(this, icinga);
+
+        this.on('layout-change', this.onLayoutChange, this);
+        this.on('rendered', '.container', this.onRendered, this);
+        this.on('click', '.collapsible + .collapsible-control', this.onControlClicked, this);
+
+        this.icinga = icinga;
+        this.defaultVisibleRows = 2;
+        this.defaultVisibleHeight = 36;
+
+        this.state = new Icinga.Storage.StorageAwareMap.withStorage(
+            Icinga.Storage.BehaviorStorage('collapsible'),
+            'expanded'
+        )
+            .on('add', this.onExternalExpansion, this)
+            .on('delete', this.onExternalCollapse, this);
+    };
+    Collapsible.prototype = new Icinga.EventListener();
```

Comment:
I would add a newline here. Else the prototype initialisation is kind of hidden.